### PR TITLE
add asm::nop, asm::delay, and delay.rs.

### PR DESCRIPTION
**Pull request by dkhayes117**
update assembly binaries.
update CHANGELOG.md

### CODE REVIEW

File: src/asm.rs [hunk 1]
```diff
@@ -73,3 +78,41 @@ pub unsafe fn sfence_vma(asid: usize, addr: usize) {
...
+pub unsafe fn delay(cycles: u32) {
+    match () {
+        #[cfg(all(riscv, feature = "inline-asm"))]
+        () => {
+            let real_cyc = 1 + cycles / 2;
+            asm!(
+            "1:",
+            "addi {0}, {0}, -1",
+            "bne {0}, zero, 1b",
+            in(reg)
```
Comment (Disasm) [inline src/asm.rs:102]: What is `reg` here? I guess it should be `real_cyc`

**Review by almindor — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Add enums `Range`, `Permission` for PMP configuration
 - Add `set_pmp()` and `clear_pmp()` functions to pmpcfg(x) modules
 - Add struct `Pmpcsr` and is returned from `pmpcfgx::read()`
+- Add delay structure and methods using embedded-hal traits and Mcycle register
+- add asm::delay function for assembly-based busy-loops
+- add asm::nop, a wrapper for implementing a `nop` instruction
```
Comment (Disasm) [inline CHANGELOG.md:16]:
```suggestion
- Add `asm::delay()` function for assembly-based busy-loops
- Add `asm::nop()`, a wrapper for implementing a `nop` instruction
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Add enums `Range`, `Permission` for PMP configuration
 - Add `set_pmp()` and `clear_pmp()` functions to pmpcfg(x) modules
 - Add struct `Pmpcsr` and is returned from `pmpcfgx::read()`
+- Add delay structure and methods using embedded-hal traits and Mcycle register
```
Comment (Disasm) [inline CHANGELOG.md:14]:
```suggestion
- Add delay structure and methods using embedded-hal traits and `mcycle` register
```

**Review by Disasm — APPROVED** [review]
Thank you!
bors r+

**Pull request merged**